src/client.ts

```diff
@@ -352,6 +362,23 @@ export class SubscriptionClient {
     }
   }
 
+  private clearInactivityTimeout() {
```

Comment:
Should there be a call to this if the connection is closed for any other reason, otherwise I could see the reconnect logic fighting with the inactivity timeout and possibly tripping the flags for `closedByUser`.

Reply:
I agree, if a connection is closed (say, manually) but inactivity timeout had started. It might close the connection after `reconnection`.

To get around this, I called `clearInactivityTimeout` inside the `close` function.